internal/runner/cypress.go

```diff
@@ -0,0 +1,92 @@
+package runner
+
+import (
+	"fmt"
+	"os/exec"
+	"slices"
+	"strings"
+
+	"github.com/buildkite/test-engine-client/internal/debug"
+	"github.com/buildkite/test-engine-client/internal/plan"
+	"github.com/kballard/go-shellquote"
+)
+
+var _ = TestRunner(Cypress{})
+
+type Cypress struct {
+	RunnerConfig
+}
+
+func (c Cypress) Name() string {
+	return "Cypress"
+}
+
+func NewCypress(c RunnerConfig) Cypress {
+	if c.TestCommand == "" {
+		c.TestCommand = "cypress run --spec {{testExamples}}"
+	}
+
+	if c.TestFileExcludePattern == "" {
+		c.TestFileExcludePattern = "**/node_modules"
+	}
+
+	if c.TestFilePattern == "" {
+		c.TestFilePattern = "**/*.cy.{js,jsx,ts,tsx}"
+	}
+
+	return Cypress{c}
+}
+
+func (c Cypress) Run(testCases []string, retry bool) (RunResult, error) {
+	cmdName, cmdArgs, err := c.commandNameAndArgs(c.TestCommand, testCases)
+	if err != nil {
+		return RunResult{Status: RunStatusError}, fmt.Errorf("failed to build command: %w", err)
+	}
+
+	cmd := exec.Command(cmdName, cmdArgs...)
+
+	fmt.Printf("%s %s\n", cmdName, strings.Join(cmdArgs, " "))
+	err = runAndForwardSignal(cmd)
+
+	if err == nil { // note: returning success early
```

Comment:
Why return success early here? The other way around (https://google.github.io/styleguide/go/decisions#indent-error-flow) would be more conventional.

Reply:
Ah, good point! In other runners, we handle the error differently and returning early is more intuitive. But not in this case, so yes we should do the other way here.